src/library/udp_proxy.cpp

```diff
@@ -165,6 +167,7 @@ void ProxyClient::HandleUdpRx(const coap::Request &aUdpRx)
 
     mEndpoint.SetPeerAddr(peerAddr);
     mEndpoint.SetPeerPort(peerPort);
+    mEndpoint.SetSockPort(peerPort);
```

Comment:
Check that the `udpEncap` message is always setting the destination port to `kDefaultMmPort`? This is the simplest fix for now:

```
peerPort = utils::Decode<uint16_t>(udpEncap->GetValue());
destPort = utils::Decode<uint16_t>(&udpEncap->GetValue()[2], 2);
VerifyOrExit(destPort == kDefaultMmPort, ERROR_UNIMPLEMENTED("dropping UDP_RX.ntf message to port {}: support only port {}", destPort, kDefaultMmPort));
```